middle_end/flambda2/from_lambda/lambda_to_flambda.ml

```diff
@@ -1489,7 +1489,14 @@ and cps_function env ~fid ~(recursive : Recursive.t) ?precomputed_free_idents
       when tag = Obj.double_array_tag ->
       assert (
         List.for_all
-          (fun kind -> kind = Lambda.(Pboxedfloatval Pfloat64))
+          (fun (kind : Lambda.value_kind) ->
+            match kind with
+            | Pboxedfloatval Pfloat64 -> true
+            | Pboxedfloatval Pfloat32
+            (* CR mshinwell: should this unboxing apply for Pfloat32? *)
```

Comment:
This case specifically detects unboxed float records. Unless the frontend also magically unboxes records of `Float32.t` into flat float arrays, the `Pfloat32` case should not happen.

Reply:
Ah yes, indeed